distributed/worker.py

```diff
@@ -1649,10 +1647,11 @@ def _close(executor, wait):
         await self.rpc.close()
```

Comment:
My gut feeling would tell me to put listener stops, RPC close and stream close together. This would basically mark the point where we're severing all outside communication and will only do some internal housekeeping afterwards. 
As it is right now, these events are still spread out a log, aren't they?